client.go

```diff
@@ -76,7 +76,7 @@ func (m *Client) listenRead() {
 			return
 		} else if err != nil {
 			m.sv.OnError(err)
-		} else {
+		} else if msg != nil {
```

Comment:
@pengyudu I think we should better to separate the blocks that error handling and empty message checking in regard of logically like below.

```go
// This block is error handling
if err == io.EOF {
    // ...
} else if err != nil {
    // ...
}

// This block is empty message checking
if msg != nil {
    // ...
}
```

How about you?

Reply:
Well, I think use this method